src/views/ListOfBooks.jsx

```diff
@@ -35,37 +37,51 @@ const ListOfBooks = () => {
     });
   },[]);
 
-  // Filter books by subject
-  const subjects = books?.flatMap((book) => book.subjects?.map((subject) => subject.name));
-
   const onlyUnique = (value, index, self) => {
     return self.indexOf(value) === index;
   }
 
+  // Filter books by subject
+  const subjects = books?.flatMap((book) => book.subjects?.map((subject) => subject.name));
   // Get unique subjects to display in the select option
   const subjectsList = subjects?.filter(onlyUnique);
 
+  // Filter books by levels
+  const levels = books?.flatMap((book) => book.levels?.map((level) => level.name));
+  // Get unique levels to display in the select option
+  const levelsList = levels?.filter(onlyUnique);
+
   return (
-   <div className="books-list">
+   <div className="book-list-page">
      <SubjectFilter
         subjectsList={subjectsList}
         activeSubject={activeSubject}
         setActiveSubject={setActiveSubject}
       />
-      {books && books.map((book) => {
-        if (activeSubject === "" || book.subjects.some((subject) => subject.name === activeSubject)) {
-          return (
-            <div className="books-container" key={book.id}>
-              <Link to={book.valid && `/book/${book.id}`} state={{ book }}>
-                <h3 className={book.valid ? "" : "not-valid-books-title"}>{book.displayTitle}</h3>
-                <img className={book.valid ? "valid-books-img" : "not-valid-books-img"} src={book.url} alt={book.displayTitle} />
-              </Link>
-            </div>
-          )
-      } else {
-        return null;
-      }})}
-   </div>
+     <LevelFilter
+        levelsList={levelsList}
+        activeLevel={activeLevel}
+        setActiveLevel={setActiveLevel}
+      />
+      <div className="books-list">
+        {books && books.map((book) => {
+          if (activeLevel === "" || book.levels.some((level) => level.name === activeLevel)) {
```

Comment:
One of these branches doesn't return (not a huge problem in Javascript, but a potential source of bugs later on).

How can you rewrite it to be safer?

Reply:
I removed the return null cause it was not necessary